docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,177 @@
+the controller will consider that a trigger to cancel the installation.
+
+Currently, there is a webhook that prevents the cluster reference from being removed
+when a host is in the middle of installation. The proposal for this enhancement is to remove
+that webhook to unblock the removal of the cluster reference.
+
+#### Cancelling Installation
+
+The host should transition to cancelled state.
+
+The `stop-installation` command is sent to a host that is in the middle of installing.
+The command will be sent with retries in case there is a scenario such as the host just starting rebooting.
+
+However, if the command doesn't receive a successful reply, the controller will assume
+the host is unreachable and transition the host to a pending-user-action state.
+
+#### Post Cancellation
+
+##### Successful Cancellation
+
+If a host cancellation was successful, the host will be transitioned to unbinding then reclaiming.
+
+During reclaim, the host should be rebooted with the discovery ISO and its state should be reset to
+discovering once the host tries to register again. This already exists with the current reclaim flow
+
+##### Failed Cancellation
+
+If the host cancellation fails due to the host being unreachable or timing out, then the
+host will transition to a pending-user-action state.
+
+From here, the user must take action by deleting the Agent CR to "reset" the host
+and rebooting the host manually from the discovery ISO.
+
+### Risks and Mitigations
+
+#### Risk 1: Unintended consequences
+
+Host installation cancellation is a completely new flow that could have unintended consequences.
+For example, it might not be able to be cancelled during a certain part of installation, or the
+host might end up in a state that's unreachable rendering it stuck.
+
+##### Mitigation
+
+The way to mitigate this is to fully explore and ensure all paths into and out of this state are covered.
+When testing out this feature, we should test cancelling the installation at different times and phases
+to ensure that cancellation is possible.
+
+## Design Details [optional]
+
+### Open Questions
+
+1. Are there limitations when cancelling an in-progress OpenShift install?
+
+### UI Impact
+
+The UI could expose a cancel install button if a host is being added to an imported or installed cluster. 
+This cancel button should remove the cluster deployment reference from the Agent to trigger a
+cancellation. The cancel button should only be exposed once the host starts installing. 
+
+### Test Plan
+
+An e2e test will need to be added or updated for day-2 host installation and cancellation.
+
+## Drawbacks
+
+There are a lot of risks when changing the host stages like this and when interrupting an installation as it can leave the host in a broken state.
+
+If the host becomes unreachable during installation and the user wants to cancel it, there may be no way of even bringing this host back in the first place and would require user intervention.
```

Comment:
Would be great if we can document the manual steps to do that.

Reply:
Great suggestion, will add that too. Thank you!